src/parsers/manifest/dash/indexes/get_index_segment.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ISegment } from "../../../../manifest";
+
+/**
+ * Construct index segment for the given index.
+ * Returns null if the index segment is on the init segment.
+ * @param {Object} index
+ * @returns {Object}
+ */
+export default function getIndexSegment(
+  index: { timescale: number;
+           representationIndex?: { media: string;
+                                  range?: [number, number]; } |
+                                 null;
+           indexRange?: [number, number];
+           indexTimeOffset : number; }
+) : ISegment|null {
+  const { representationIndex } = index;
+  if (representationIndex == null) {
+    return null;
+  }
+  return { id: "representationIndex",
+           isInit: false,
+           noMediaData: true,
```

Comment:
I find it still a little strange to name `noMediaData` something that every body knows in the code to be an index segment. Why not just `isIndex`/`isIndexSegment`?

Reply:
I got a problem with the fact that `isIndex` means that there is an index range, but does not say that there is no media data, or that it is not an init. In the pipelines, we just need to know that we don't want to parse / get data from the segment.

I think it is not sufficient to test if a segment `isIndex` and `!isInit` as an alternative.